lib/diff_file.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+class rex_ydeploy_diff_file
+{
+    private $create = [];
+    private $alter = [];
+    private $drop = [];
+
+    public function createTable(rex_sql_table $table)
+    {
+        $this->create[] = $table;
+    }
+
+    public function dropTable($tableName)
+    {
+        $this->drop[$tableName] = true;
+    }
+
+    public function ensureColumn($tableName, rex_sql_column $column)
+    {
+        $this->alter[$tableName]['ensure'][] = $column;
+    }
+
+    public function removeColumn($tableName, $columnName)
+    {
+        $this->alter[$tableName]['remove'][] = $columnName;
+    }
+
+    public function setPrimaryKey($tableName, array $primaryKey)
+    {
+        $this->alter[$tableName]['primaryKey'] = $primaryKey;
+    }
+
+    public function isEmpty()
+    {
+        return !$this->create && !$this->alter && !$this->drop;
+    }
+
+    public function getContent()
+    {
+        $content = <<<'EOL'
+<?php
+
+$sql = rex_sql::factory();
+$sql->setQuery('SET FOREIGN_KEY_CHECKS = 0');
+EOL;
```

Comment:
Aktuell sind die Migrationsdateien reine PHP-Dateien wo direkt die Migrationsbefehle drin stehen.

Beispiel:
```php
<?php

$sql = rex_sql::factory();
$sql->setQuery('SET FOREIGN_KEY_CHECKS = 0');

rex_sql_table::get('rex_foo')
    ->ensureColumn(new rex_sql_column('id', 'int(11) unsigned', false, null, null))
    ->ensureColumn(new rex_sql_column('id2', 'int(11)', false, null, null))
    ->setPrimaryKey(['id', 'id2'])
    ->alter();

$sql->setQuery('SET FOREIGN_KEY_CHECKS = 1');
```

Ich überlege, ob ich es besser, ähnlich wie bei Propel, in eine Klasse einbetten soll, innerhalb einer methode `up()`.
Falls ich mal auch die Gegenrichtigung implementiere. Dann würde noch die `down()`-Methode hinzukommen.

Für den Anfang würde ich mir das gerne noch sparen. Aber vielleicht wäre es in der Zukunft schon ganz praktisch. Daher vielleicht gleich direkt in die Richtung gehen, also zumindest eben schon mit der Klasse und der `up()`-Methode.
@staabm Oder was meinst du dazu?